homeassistant/components/luftdaten/config_flow.py

```diff
@@ -77,6 +79,13 @@ def _show_form(self, errors=None):
         if not valid:
             return self._show_form({CONF_SENSOR_ID: 'invalid_sensor'})
 
+        available_sensors = [x for x in luftdaten.values
+                             if luftdaten.values[x] is not None]
```

Comment:
What is the shape of `luftdaten.values`? Because Python will throw if a key does not exist.

Reply:
It is initialized in the luftdaten library, in `__init__` of the Luftdaten object:
```
self.values = {
                'humidity': None,
                'P1': None,
                'P2': None,
                'pressure': None,
                'temperature': None,
}
```

Reply:
I checked a half a dozens stations and it seems that the Luftdaten API is no longer providing humidity, pressure and temperature combined with the pollution levels. It looks like that there are now two types of stations:

- P1 and P2
- humidity and temperature (and sometimes pressure)

The proper fix would be to limit `python-luftdaten` to P1 and P2. Especially in regard that this should become an `air_quality` integration.

Reply:
True, most physical devices have two sensors, one particle and one for humidity/temp/pressure. Both get an unique id. So a single id will never have all sensors. I agree the P1 and P2 measurement should become an `air_quality` integration. However, in the short term I think this PR will make the luftdaten integration easier and better than than the yaml config.